chainer/dataset/tabular/from_data.py

```diff
@@ -0,0 +1,111 @@
+import chainer
+from chainer.dataset.tabular import tabular_dataset
+
+
+def from_data(*args, **kwargs):
+    """Create a :class:`~chainer.dataset.TabularDataset` from lists/arrays.
+
+    Args:
+        args (list or array or tuple): Data of a column.
+            If this argument is an array or a list,
+            it is treated as :obj:`data`.
+            In this case, the key is generated automatically.
+            Do not rely on the key.
+            If this argument is a tuple, it is treated as :obj:`(key, data)`.
+        kwargs (list or array): Data of a column.
+            The order of columns is determined automatically.
+            Do not rely on the order.
+    Return:
+        A :class:`~chainer.dataset.TabularDataset`.
+        If only one argument is given, :attr:`mode` is :obj:`None`.
+        If more than one arguments are given as :obj:`args`,
+        :attr:`mode` is :class:`tuple`.
+        If more than one arguments are given as :obj:`kwargs`,
+        :attr:`mode` is :class:`dict`.
```

Comment:
Can we add some example code? The specifications for 1) singleton args, 2) multiple args, 3) kwargs are a little bit complicated, so it's nice to have simple examples for them.